public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/utils.ts

```diff
@@ -14,3 +16,21 @@ export function parseResourceURI(resourceURI: string) {
 export function isGUIDish(input: string) {
   return !!input.match(/^[A-Z0-9]+/i);
 }
+
+export function findRow(rows: ResourceRowGroup, id: string): ResourceRow | undefined {
+  for (const row of rows) {
+    if (row.id === id) {
+      return row;
+    }
+
+    if (row.children) {
+      const result = findRow(row.children, id);
+
+      if (result) {
+        return result;
+      }
+    }
+  }
+
+  return undefined;
```

Comment:
this seems like it can be easier using `findBy` of lodash? smth like: 
```js
return findBy(rows, r => r.id === id || (row.children && findRow(row.children, id));
```
 anyway, both are simple enough

Reply:
Yeah I used something like this originally (Array.prototype.find actually), but the problem with find and findBy is that they don't work for nested items. While it will traverse children, it will still return the top-level row that contains a leaf row. We want to return the leaf row itself.

Definitely open to something cleaner though!